frontend/src/components/Form/SearchableSelect.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+
+import Label from './Label';
+
+import styles from './searchableSelect.cssmodule.scss';
+
+const SearchableSelect = ({
+  label,
+  hint,
+  type,
+  required,
+  field,
+  options,
+  onChange,
+  hiddenOption,
+  value,
+  ...remainingProps
+}) => {
+  const inputOption = {
+    ...remainingProps,
+    className: 'fr-select midlength-input',
+    id: field,
+    type,
+    required,
+    value,
+    onChange: e => setInternalValue(e.target.value),
+  };
+
+  const [internalValue, setInternalValue] = useState(value);
+  const [showOptions, setShowOptions] = useState(false);
+
+  const onInternalChange = (id, newValue) => {
+    setInternalValue(newValue);
+    onChange(id, field);
+  };
+
+  const onFocus = () => {
+    onInternalChange('', '');
+    setShowOptions(true);
+  };
+
+  const onBlur = () => {
+    const foundValue = options.find(option => option.label === internalValue);
+    if (!foundValue) {
+      setInternalValue('');
+    }
+    setShowOptions(false);
+  };
+
+  const filteredOptions = options.filter(option => option.label.toLowerCase().includes(internalValue.toLowerCase()));
```

Comment:
I would extract the compare/filter function to have proper test on it. It would be easier to extend. We could for example handle special chars, spaces, etc. We may already have something like this elsewhere.